go/private/rules/stdlib.bzl

```diff
@@ -16,17 +16,107 @@ load("@io_bazel_rules_go//go/private:providers.bzl",
     "GoStdLib",
 )
 
+STDLIB_GOOS_GOARCH = [
+    ("darwin", "amd64"),
+    ("linux", "386"),
+    ("linux", "amd64"),
+    ("linux", "armv6l"),
+    ("windows", "386"),
+    ("windows", "amd64"),
+    ("freebsd", "386"),
+    ("freebsd", "amd64"),
+    ("linux", "ppc64le"),
+    ("linux", "s390x"),
+]
+
+_STDLIB_BUILD = """
+load("@io_bazel_rules_go//go/private:rules/stdlib.bzl", "stdlib")
+
+stdlib(
+    name = "{name}",
+    goos = "{goos}",
+    goarch = "{goarch}",
+    race = {race},
+    cgo = {cgo},
+    visibility = ["//visibility:public"],
+)
+"""
+
 def _stdlib_impl(ctx):
+  go = ctx.actions.declare_file("bin/go") # TODO: .exe
+  src = ctx.actions.declare_directory("src")
+  pkg = ctx.actions.declare_directory("pkg")
+  root_file = ctx.actions.declare_file("ROOT")
+  files = [root_file, go, src, pkg]
+  goroot = root_file.path[:-(len(root_file.basename)+1)]
+  sdk = ""
+  for f in ctx.files._host_sdk:
+    prefix, found, _  = f.path.partition("bin/go")
+    if found:
+      sdk = prefix
+  if not sdk:
+    fail("Could not find go executable in go_sdk")
+  cpp = ctx.fragments.cpp
+  features = ctx.features
+  options = (cpp.compiler_options(features) +
+        cpp.unfiltered_compiler_options(features) +
+        cpp.link_options +
+        cpp.mostly_static_link_options(features, False))
+  linker_path, _ = cpp.ld_executable.rsplit("/", 1)
+  ctx.actions.write(root_file, "")
+  cc_path = cpp.compiler_executable
+  if not cpp.compiler_executable.startswith("/"):
```

Comment:
When is this true?

Reply:
When it's a relative path to a thing in another repository, specifically external/local_config_cc/cc_wrapper.sh on mac *sigh*